cmd/ssh_for_alicloud.go

```diff
@@ -0,0 +1,576 @@
+	checkError(err)
+	a.ZoneID, err = decodedQuery.String("ZoneId")
+	checkError(err)
+	a.VpcID, err = decodedQuery.String("VpcAttributes", "VpcId")
+	checkError(err)
+	a.VSwitchID, err = decodedQuery.String("VpcAttributes", "VSwitchId")
+	checkError(err)
+	a.ImageID, err = decodedQuery.String("ImageId")
+	checkError(err)
+	a.ShootName = getShootClusterName()
+	a.BastionSecurityGroupName = a.ShootName + "-bsg"
+	a.BastionInstanceName = a.ShootName + "-bastion"
+
+	a.InstanceChargeType = "PostPaid"
+	a.InternetChargeType = "PayByTraffic"
+	a.InternetMaxBandwidthIn = "10"
+	a.InternetMaxBandwidthOut = "100"
+	a.IoOptimized = "optimized"
+	a.KeyPairName = a.ShootName + "-ssh-publickey"
+	a.InstanceType = a.getMinimumInstanceSpec()
+}
+
+// createBastionHostSecurityGroup finds the or creates a security group for the bastion host.
+func (a *AliyunInstanceAttribute) createBastionHostSecurityGroup() {
+	res, err := ExecCmdReturnOutput("bash", "-c", "aliyun ecs DescribeSecurityGroups --VpcId="+a.VpcID)
+	checkError(err)
+	decodedQuery := decodeAndQueryFromJSONString(res)
+
+	securityGroupNames, err := decodedQuery.Array("SecurityGroups", "SecurityGroup")
+	checkError(err)
+	checkSecurityGroupExists := false
+	for _, iter := range securityGroupNames {
+		securityGroup := jsonq.NewQuery(iter)
+		name, err := securityGroup.String("SecurityGroupName")
+		checkError(err)
+		if name == a.BastionSecurityGroupName {
+			checkSecurityGroupExists = true
+			a.BastionSecurityGroupID, err = securityGroup.String("SecurityGroupId")
+			checkError(err)
+		}
+	}
+
+	if checkSecurityGroupExists == false {
+		res, err = ExecCmdReturnOutput("bash", "-c", "aliyun ecs CreateSecurityGroup --RegionId="+a.RegionID+" --VpcId="+a.VpcID+" --SecurityGroupName="+a.BastionSecurityGroupName)
+		checkError(err)
+		decodedQuery = decodeAndQueryFromJSONString(res)
+		a.BastionSecurityGroupID, err = decodedQuery.String("SecurityGroupId")
+		checkError(err)
+		attemptCnt := 0
+		for attemptCnt < 60 {
+			res, err = ExecCmdReturnOutput("bash", "-c", "aliyun ecs DescribeSecurityGroups --SecurityGroupIds=\"['"+a.BastionSecurityGroupID+"']\"")
+			checkError(err)
+			decodedQuery = decodeAndQueryFromJSONString(res)
+			totalCount, err := decodedQuery.Int("TotalCount")
+			checkError(err)
+			if totalCount == 1 {
+				time.Sleep(time.Second * 30)
+				fmt.Println("Bastion host security group created.")
+				break
+			}
+			fmt.Println("Creating bastion host security group...")
+			time.Sleep(time.Second * 2)
+			attemptCnt++
+		}
+		if attemptCnt == 60 {
+			fmt.Println("Bastion host security group creation time out. Please try again.")
+			os.Exit(2)
+		}
+		fmt.Println("Configuring bastion host security group rules...")
+		res, err = ExecCmdReturnOutput("bash", "-c", "aliyun ecs AuthorizeSecurityGroup --Policy Accept --NicType intranet --Priority 1 --SourceCidrIp 0.0.0.0/0 --PortRange 22/22 --IpProtocol udp --SecurityGroupId="+a.BastionSecurityGroupID)
```

Comment:
Just as comment:
I used these commands to setup a bastion host manually.
With the command in this line I got an error "unknown region" or similar. 
Resolution was to add a `--RegionID` argument in my case.

Reply:
Hi Martin,
Which region are you running aliyuncli in? Is it the same region where your security group is created?

Reply:
Also, just a reminder, don't forget to delete the security groups & bastion host instance manually created. If you don't delete them, shoot deletion will fail since terraform configuration is changed.